src/showingpreviously/cinemas/centre_for_the_moving_image.py

```diff
@@ -0,0 +1,134 @@
+    film = Film(name=name, year=release_year)
+    for showing_time in showing_times.find_all('a', class_='btn-times'):
+        showing_time_str = showing_time.text.strip().split('\n')[0].strip()
+        showing_datetime = datetime.datetime.strptime(f'{showing_date_str} {showing_time_str}', '%Y-%m-%d %H:%M')
+        showing_screen = Screen(name=showing_time.find('span', class_='screen').text.strip())
+        showing_type_attributes = get_showing_type_attributes(name, showing_date_str, showing_time)
+        showing = Showing(
+            film = film,
+            time = showing_datetime,
+            chain = CHAIN,
+            cinema = cinema['cinema'],
+            screen = showing_screen,
+            json_attributes = {**attributes, **showing_type_attributes}
+        )
+        showing_list.append(showing)
+    return showing_list
+
+
+def get_showing_type_attributes(film_name: str, showing_date_str: str, showing: BeautifulSoup) -> dict[str, any]:
+    showing_type_list = showing.find_all('span', class_='showing-type')
+    attributes = {}
+    for showing_type in showing_type_list:
+        showing_type_attribute = [attr for attr in showing_type['class'] if attr != 'showing-type']
+        if len(showing_type_attribute) != 1:
+            raise CinemaArchiverException('Expected one showing type attribute for {film_name} on {showing_date_str}: {str(showing_type_attribute)}')
+        showing_type_attribute = showing_type_attribute[0]
+        if showing_type_attribute not in KNOWN_SHOWING_TYPE_ATTRIBUTES:
+            raise CinemaArchiverException('Unknown showing type attribute for {film_name} on {showing_date_str}: {showing_type_attribute}')
+        attributes[showing_type_attribute] = True
+    return attributes
+
+
+def get_film_page_details(cinema: dict[str, any], film_url: str) -> (str, dict[str, any]):
+    global film_page_details_cache
+    if not film_url.startswith('http'):
+        film_url = f'{cinema["base_url"]}{film_url}'
+    if film_url in film_page_details_cache:
+        film_details = film_page_details_cache[film_url]
+        return film_details
+    req = get_response(film_url)
+    soup = BeautifulSoup(req.text, features='html.parser')
+    page_title = soup.find('title').text
+    if 'Access Denied' in page_title:
+        return ('', {})
+    attributes = {}
+    details_list = soup.find('div', class_='event-detail')
+    release_year = get_specific_film_attribute(details_list, 'release_year')
+    attributes['language'] = get_specific_film_attribute(details_list, 'languages')
+    attributes['format'] = get_specific_film_attribute(details_list, 'format')
+    film_page_details_cache[film_url] = (release_year, attributes)
+    return (release_year, attributes)
+
+
+def get_specific_film_attribute(details_list: BeautifulSoup, attribute_name: str) -> str:
+    attribute_soup = details_list.find('li', class_=attribute_name)
+    if attribute_soup == None:
```

Comment:
Better to have this with the `is None` identity operator, rather than `== None` equality operator.